tools/analyzer_plugin/test/unit/util/boilerplate_utils_test.dart

```diff
@@ -84,13 +80,13 @@ void main() {
     Future<void> checkPartValidity({
       bool shouldBeValid = true,
       bool isPartOverReact = true,
-      @required String path,
+      required String path,
     }) async {
       final result = await parseAndGetResolvedUnit(
         isPartOverReact ? sourceWithOverReactPart : sourceWithNonOverReactPart,
         path: path,
       );
-      final part = result.unit.directives.whereType<PartDirective>().firstOrNull;
+      final part = result.unit.directives.whereType<PartDirective>().firstOrNull!;
```

Comment:
`whereType` already results in an `Iterable<PartDirective>`, so `whereNotNull` here wouldn't have any affect.

For doing it some other "safer" way, since we're expecting a part to exist here, there's really not a "safe" way to do this unless we were to perform a custom null-check and throw some other sort of error.

Doing `.firstOrNull!` is a bit silly though, and this could instead be:
```suggestion
      final part = result.unit.directives.whereType<PartDirective>().first;
```
or really perhaps
```suggestion
      final part = result.unit.directives.whereType<PartDirective>().single;
```
and we could remove the `expect(part, isNotNull);` below.

I can clean this up.